### PR TITLE
incorporate spec changes

### PR DESCRIPTION
**Pull request by mandelsoft**
- new access method for helm repositories
- component descriptor normalization according to RFC8785 (JCS)
- version mapping for mapping OCM versions to OCI tags.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: doc/glossary.md [hunk 1]
```diff
@@ -106,6 +106,14 @@ behaviour of such elements.
 additional parts of the identity of an [element](#elemid) of a [component version](#compvers)
 described as dedicated attributes in a [component descriptor](#compdesc).
 
+## H
+
+#### [Helm Chart](appendix/B/helm.md)<a id="helmchart"/>
+An set of files describing the deplyoment of a Kubernetes application using
```
Comment (jensh007) [inline doc/glossary.md:112]: Typo: A set

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 2]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+
+- *extraction of the signature relevant information from the component descriptor*
+
+  The result is basically a JSON object, which decsribed the relevant information.
+
+- *normalization of the resulting JSON object*
+
+  Here, the object is serialized to a unique and reproducable byte sequence, which is finally used to determine the digest.
+
+  There are to such normalizations:
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:35]: type: Two

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 3]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+# `jsonNormalisationV1` vs `jsonNormalisationV2`
+
+The `JsonNormalisationV1` serialization format is based on the serialization
+format of the component descriptor. It uses an appropriate JSON object containing the relevant fields as contained in the component descriptors's serialization. The format version fields are included. Therefore, the normalized form is depending on the chosen serialization format. Changing this format version would result in different digests. The resulting JSON object is serialized with the [OCM specific scheme](#generic-normalization-format)
+
+`JsonNormalisationV2` strictly uses only the relevant component descriptor
+information according to the field specification shown below. It is independent of the serialization format used to store the component decsriptor in some storage backend. Therefore, the calculated digest in finally independent of the serialization format chosen for storing the component descriptor in a storage backend.
+
+  Additionally, it uses the JCS scheme for uniquely 
+  serialize the resulting JSON object.
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:49]: typo: serializing

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 4]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+
+## Relevant information in Component Descriptors
+
+A component descriptor contains signature relevant information and
+information, which may change. For example, the access methods specifications
+might be changed during atransport step.
+
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:56]: Suggestion: A component descriptor contains static information and information which may change (for example the access methods specifications usually will be changed during a transport step. A digest should be stable even after transport and therefore should only hash static information.

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 5]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+### Access Methods
+
+Access method specifications are completely ignored.
+A resource/source is ignored, if the access method type is `none`.
+
+## Labels
+
+Labels are removed before signing but can be marked with a special boolean
+property `signing` not to be removed and thus be part of the signature.
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:74]: Labels are removed before signing but can be marked with a special boolean property `signing`. This property indicates that the label should be part of the digest.

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 6]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+labels:
+- name: label1
+  value: foo
+- name: label2
+  value: bar
+  signing: true
+```
+
+`label1` will be excluded from the signature, `label2` will be included.
+The label values is takes as it is, preserving a potentially deep structure.
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:90]: signature --> digest
typo: taken
deep --> deeply nested

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 7]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+
+## Exclude Resource from Normalisation/Signing
+
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:93]: Normalization

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 8]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+If a resource should not be part of the normalisation and later signing, the resource needs a special digest in the following format:
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:94]: normalization

File: doc/specification/formats/componentdescriptor_normalization.md [hunk 9]
```diff
@@ -22,7 +22,85 @@ normalizations.
...
+
+```yaml
+digest:
+  hashAlgorithm: NO-DIGEST
+  normalisationAlgorithm: EXCLUDE-FROM-SIGNATURE
```
Comment (jensh007) [inline doc/specification/formats/componentdescriptor_normalization.md:99]: normalizationAlgorithm

File: doc/specification/formats/normalization.md [hunk 10]
```diff
@@ -5,7 +5,7 @@ content of described artifacts must be incorporated.
 
 The metadata of a component version is described by a [component-descriptor](../../specification/elements/README.md#component-descriptor). It contains volatile information, also, e.g. the artifact access specification, which might change during [transports](../../introduction/transports.md).
 Therefore, to calculate a digest, the component descriptor has to be transformed
-to an immutable technical reprsentation containing only signature relevant information. THis process is called [*component descriptor normalization*](componentdescriptor_normalization.md).
+to an immutable technical reprsentation containing only signature relevant information. This process is called [*component descriptor normalization*](componentdescriptor_normalization.md).
```
Comment (jensh007) [inline doc/specification/formats/normalization.md:8]: representation

**Review by jensh007 — CHANGES_REQUESTED** [review]

File: doc/specification/formats/componentdescriptor_normalization.md (hunk 9 above)
Comment (mandelsoft) [reply doc/specification/formats/componentdescriptor_normalization.md:99]: Ha, unfortunaltely this is pat of the interface, which cannot be easily changed. I asked for this right after it has been introduced, but there were already lots of CV.

**Review by jensh007 — APPROVED** [review]

**Pull request merged**